test-ng: `test_nvme.py` fails on trustedboot and USI flavors but does not throw errorcode != 0

**What happened:**

`test_nvme.py` fails on trustedboot and USI flavors as it is not allowed to write to `/mnt/nvme`.

This does not throw errorcode != 0 and the overall pytest runs does not let the pipeline fail.

**Debug Logs:**

https://github.com/gardenlinux/gardenlinux/actions/runs/17998820566/job/51205418541

```
🧪  running tests with args: ./run_tests --junit-xml=/dev/virtio-ports/test_junit --system-booted --allow-system-modifications --skip-performance-metrics
================================================= test session starts ==================================================
platform linux -- Python 3.13.5, pytest-8.4.2, pluggy-1.6.0
rootdir: /run/gardenlinux-tests/tests
collected 55 items

test_basics.py ...s..                                                                                            [ 10%]
test_containers.py ..                                                                                            [ 14%]
test_network.py .......s                                                                                         [ 29%]
test_nodejs.py ss                                                                                                [ 32%]
test_nvme.py .E                                                                                                  [ 36%]
test_packages.py ......                                                                                          [ 47%]
test_proc.py ..                                                                                                  [ 50%]
test_sapmachine.py sss                                                                                           [ 56%]
test_ssh.py ..........s..                                                                                        [ 80%]
test_users.py ...                                                                                                [ 85%]
test_vmware.py ssssssss                                                                                          [100%]

======================================================== ERRORS ========================================================
_________________________________________ ERROR at setup of test_nvme_locally __________________________________________
handlers/configure_nvme.py:66: in nvme_device
    os.makedirs("/mnt/nvme")
E   OSError: [Errno 30] Read-only file system: '/mnt/nvme'
------------------------------------------------ Captured stdout setup -------------------------------------------------
connecting to device: nvme0
Discarding device blocks:      0/131072             done                            
Creating filesystem with 131072 4k blocks and 32768 inodes
Filesystem UUID: 63e9c17e-a49f-498b-a106-a9f4ad3a8950
Superblock backups stored on blocks: 
	32768, 98304

Allocating group tables: 0/4   done                            
Writing inode tables: 0/4   done                            
Creating journal (4096 blocks): done
Writing superblocks and filesystem accounting information: 0/4   done

------------------------------------------------ Captured stderr setup -------------------------------------------------
mke2fs 1.47.2 (1-Jan-2025)
...
=============================================== short test summary info ================================================
...
ERROR test_nvme.py::test_nvme_locally - OSError: [Errno 30] Read-only file system: '/mnt/nvme'
================================= 38 passed, 16 skipped, 4 warnings, 1 error in 32.53s =================================
```

--- a/.github/workflows/test_flavor_qemu.yml
+++ b/.github/workflows/test_flavor_qemu.yml
@@ -25,7 +25,7 @@ jobs:
       - name: install qemu
         run: |
           sudo apt-get update
-          sudo env DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends qemu-utils qemu-system-x86 qemu-system-arm swtpm socat
+          sudo env DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends qemu-utils qemu-system-x86 qemu-system-arm swtpm socat libxml2-utils
       - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # pin@v5.0.0
         with:
           submodules: true

--- a/tests-ng/README.md
+++ b/tests-ng/README.md
@@ -66,6 +66,7 @@ Before running the test framework, make sure the following dependencies are inst
 - `make`
 - `curl`
 - `jq`
+- `libxml2-utils`
 - `unzip`
 - `qemu`
 - `qemu-utils`
@@ -74,14 +75,14 @@ Before running the test framework, make sure the following dependencies are inst
 
 ```
 apt-get update
-apt-get install podman make curl jq unzip qemu swtpm socat
+apt-get install podman make curl jq libxml2-utils unzip qemu swtpm socat
 
 ```
 
 #### Install on MacOS
 
 ```
-brew install coreutils bash gnu-sed gnu-getopt podman make curl jq unzip swtpm socat
+brew install coreutils bash gnu-sed gnu-getopt podman make curl jq libxml2 unzip swtpm socat
 ```
 
 ### Basic Usage

--- a/tests-ng/util/run_qemu.sh
+++ b/tests-ng/util/run_qemu.sh
@@ -396,5 +396,7 @@ else
 	cat "$tmpdir/serial.log"
 fi
 
-# TODO: find a better way to check if the tests failed
-! (tail -n1 "$tmpdir/serial.log" | grep failed >/dev/null)
+num_errors=$(xmllint --xpath 'string(/testsuites/testsuite/@errors)' "$tmpdir/junit.xml")
+if [ "${num_errors}" -gt 0 ]; then
+    exit 1
+fi

